src/libs/Network/processRequest.js

```diff
@@ -11,11 +32,43 @@ import * as NetworkEvents from './NetworkEvents';
  * @returns {Promise}
  */
 export default function processRequest(request) {
+    const persisted = lodashGet(request, 'data.persist', false);
     const finalParameters = enhanceParameters(request.command, request.data);
 
     // When the request goes past a certain amount of time we trigger a re-check of the connection
     const cancelRequestTimeoutTimer = NetworkEvents.startRecheckTimeoutTimer();
-    NetworkEvents.triggerRequestMade(request, finalParameters);
+    logRequestDetails(request, finalParameters);
     return HttpUtils.xhr(request.command, finalParameters, request.type, request.shouldUseSecure)
+        .then((response) => {
+            if (persisted) {
+                PersistedRequests.remove(request);
+            }
+            NetworkEvents.triggerResponse(request, response);
```

Comment:
NAB: I was thinking if we set here `request.resolve` and `request.reject` to empty promises if the requests are coming from storage (promise handling logic stripped), something like:

```js
request.resolve = request.resolve || Promise.resolve;
request.reject = request.reject || Promise.resolve;
```

so we can handle them gracefully here (and remove the conditions to check if `request.resolve` and `request.reject`exist): 
https://github.com/Expensify/App/blob/main/src/libs/API.js#L66-L109

Reply:
So add them right here before the `triggerResponse()` call instead of checking in the handler like this?

```js
request.resolve = request.resolve || Promise.resolve; 
request.reject = request.reject || Promise.resolve;
NetworkEvents.triggerResponse(request, response);
```

I like that we could then add fewer checks in the handler. I also worry that if someone adds a new call to `triggerResponse()` they will have to remember the method expects these fields to exist.

Reply:
Yeap or at the beginning of this function. 

>I also worry that if someone adds a new call to triggerResponse() they will have to remember the method expects these fields to exist.

I think in that case the error will show up in the console log and will be a bit more identifiable

Reply:
Not gonna do this one. Thanks for the thoughts.